lib/gitHubHelper.js

```diff
@@ -0,0 +1,103 @@
+"use strict";
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+var __importStar = (this && this.__importStar) || function (mod) {
+    if (mod && mod.__esModule) return mod;
+    var result = {};
+    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
+    result["default"] = mod;
+    return result;
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+const core = __importStar(require("@actions/core"));
+const dockleHelper = __importStar(require("./dockleHelper"));
+const fileHelper = __importStar(require("./fileHelper"));
+const inputHelper = __importStar(require("./inputHelper"));
+const trivyHelper = __importStar(require("./trivyHelper"));
+const httpClient_1 = require("./httpClient");
+function createCheckRunWithScanResult(trivyStatus, dockleStatus) {
+    return __awaiter(this, void 0, void 0, function* () {
+        const checkRunUrl = `https://api.github.com/repos/${process.env.GITHUB_REPOSITORY}/check-runs`;
+        const gitHubToken = core.getInput("github-token");
+        const pullRequestHeadSha = getPullRequestHeadSha();
+        const checkConclusion = getCheckConclusion(trivyStatus, dockleStatus);

```

Comment:
So we are considering dockle failure also as failure ? Or is it only for checkrun posting , the task still passes with dockle failing ?

Reply:
I was setting check run as failed when Dockle finds violations. Made it consistent with task behaviour, i.e. only fail if Trivy scan finds vulnerabilities.